agent/command/shell.go

```diff
@@ -231,8 +231,12 @@ func (c *shellExec) Execute(ctx context.Context, _ client.Communicator, logger c
 	}
 
 	err = cmd.Run(ctx)
-
-	err = errors.Wrapf(err, "command encountered problem")
+	if !c.Background && err != nil {
+		if exitCode, _ := cmd.Wait(ctx); exitCode != 0 {
+			err = errors.Errorf("shell command encountered problem: exit code %d", exitCode)
```

Comment:
It's not necessary to include the "shell command encountered problem" part since it's already wrapped with that message below this.

Reply:
ah done